src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -3466,3 +3466,86 @@ impl Step for CodegenGCC {
         cargo.into_cmd().run(builder);
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TestFloatParse {
+    path: PathBuf,
+}
+
+impl Step for TestFloatParse {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/etc/test-float-parse")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        for path in run.paths {
+            let path = path.assert_single_path().path.clone();
+            run.builder.ensure(Self { path });
+        }
+    }
+
+    fn run(self, builder: &Builder<'_>) {
+        let bootstrap_host = builder.config.build;
+        let compiler = builder.compiler(0, bootstrap_host);
+        let path = self.path.to_str().unwrap();
+        let crate_name = self.path.components().last().unwrap().as_os_str().to_str().unwrap();
+
+        // Run any unit tests in the crate
+        let cargo_test = tool::prepare_tool_cargo(
+            builder,
+            compiler,
+            // todo: seems like this should be `Mode::Std`
+            // Mode::Std,
+            Mode::ToolBootstrap,
+            bootstrap_host,
+            "test",
+            path,
+            SourceType::InTree,
+            &[],
+        );
+
+        run_cargo_test(
+            cargo_test,
+            &[],
+            &[],
+            crate_name,
+            crate_name,
+            compiler,
+            bootstrap_host,
+            builder,
+        );
+
+        // Run the actual parse tests.
+        // TODO: can we ensure this always runs with `--release`? Running without it is way
```

Comment:
I don't think so, but in general optimizations are basically always on for everything compiled by bootstrap (https://github.com/rust-lang/rust/blob/366e558c3be8936bdbf3f733cb5eb48a4feb2ae4/src/bootstrap/src/core/builder.rs#L1405-L1411) and I'd not expect that to change soon. I don't think a TODO here is useful, the same can be said for basically all of our tools (e.g., the compiler).

Reply:
Okay, great. I have dependencies as opt-level=3 anyway even for debug builds, which helps. 

Yeah the TODO was just to make sure it came up in review, removed.